src/Liquid/Tag/TagCase.php

```diff
@@ -99,14 +99,13 @@ public function endTag()
 	 */
 	public function unknownTag($tag, $params, array $tokens)
 	{
-		$whenSyntaxRegexp = new Regexp('/' . Liquid::get('QUOTED_FRAGMENT') . '/');
-
 		switch ($tag) {
 			case 'when':
+				$whenSyntax = preg_match_all('/(?<=,|or|^)\s*((["\'])?(?(2)(?:(?!\2).)*+\2|\b\w+))/', $params, $matches, PREG_UNMATCHED_AS_NULL);
```

Comment:
What is the reason we're not using `Liquid::get('QUOTED_FRAGMENT')` here? What if someone uses a different regex for `QUOTED_FRAGMENT`?

Reply:
I put that one back in.